src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -2846,7 +2847,8 @@ MergeTreeData::MutableDataPartsVector MergeTreeData::tryLoadPartsToAttach(const
     for (const auto & part_names : renamed_parts.old_and_new_names)
     {
         LOG_DEBUG(log, "Checking part " << part_names.second);
-        MutableDataPartPtr part = createPart(part_names.first, name_to_disk[part_names.first], source_dir + part_names.second);
+        auto single_disk_volume = std::make_shared<SingleDiskVolume>("volume_" + part_names.first, name_to_disk[part_names.first]);
```

Comment:
I think we should create something like `TemporarySingleDiskVolume` or `VolumeForPart`.

Reply:
You mean just rename SingleDiskVolume to something else? Why this one is bad?